src/main/scala/cromwell/webservice/CromwellApiService.scala

```diff
@@ -42,62 +45,74 @@ trait CromwellApiService extends HttpService with PerRequestCreator {
+    }
+  }
+
   def queryRoute =
     path("workflows" / Segment / Segment / "status") { (version, id) =>
-      get {
-        Try(WorkflowId.fromString(id)) match {
-          case Success(workflowId) =>
-            requestContext => perRequest(requestContext, CromwellApiHandler.props(workflowManager), CromwellApiHandler.WorkflowStatus(workflowId))
-          case Failure(ex) =>
-            complete(StatusCodes.BadRequest)
+      traceName("status") {
+        get {
+          Try(WorkflowId.fromString(id)) match {
+            case Success(workflowId) =>
+              requestContext => perRequest(requestContext, CromwellApiHandler.props(workflowManager), CromwellApiHandler.WorkflowStatus(workflowId))
+            case Failure(ex) =>
+              complete(StatusCodes.BadRequest)
+          }
         }
       }
     }
 
   def abortRoute =
     path("workflows" / Segment / Segment / "abort") { (version, id) =>
-      post {
-        Try(WorkflowId.fromString(id)) match {
-          case Success(workflowId) =>
-            requestContext => perRequest(requestContext, CromwellApiHandler.props(workflowManager), CromwellApiHandler.WorkflowAbort(workflowId))
-          case Failure(ex) =>
-            complete(StatusCodes.BadRequest)
+      traceName("abort") {
+        post {
+          Try(WorkflowId.fromString(id)) match {
+            case Success(workflowId) =>
+              requestContext => perRequest(requestContext, CromwellApiHandler.props(workflowManager), CromwellApiHandler.WorkflowAbort(workflowId))
+            case Failure(ex) =>
+              complete(StatusCodes.BadRequest)
+          }
         }
       }
     }
 
   def submitRoute =
     path("workflows" / Segment) { version =>
-      post {
-        formFields("wdlSource", "workflowInputs".?, "workflowOptions".?) { (wdlSource, workflowInputs, workflowOptions) =>
-          val tryInputsMap = Try(workflowInputs.getOrElse("{}").parseJson)
-          val tryOptionsMap = Try(workflowOptions.getOrElse("{}").parseJson)
-          (tryInputsMap, tryOptionsMap) match {
-            case (Success(JsObject(_)), Success(options: JsObject)) =>
-              if (!options.fields.values.forall(_.isInstanceOf[JsString])) {
-                complete(StatusCodes.BadRequest, "Workflow options must be a string -> string map")
-              }
-              else {
-                WorkflowOptions.fromJsonObject(options) match {
-                  case Success(wfOptions) =>
-                    requestContext => perRequest(
-                      requestContext,
-                      CromwellApiHandler.props(workflowManager),
-                      CromwellApiHandler.WorkflowSubmit(
-                        WorkflowSourceFiles(
-                          wdlSource, workflowInputs.getOrElse("{}"), wfOptions.asPrettyJson
+      traceName("submit") {
+        post {
+          formFields("wdlSource", "workflowInputs".?, "workflowOptions".?) { (wdlSource, workflowInputs, workflowOptions) =>
```

Comment:
obviously preexisting, but why is all this logic all the way out at the endpoint?